wix/stylable
# refactor(core): css-keyframes feature extraction

## fix
- [x] unresolved imported keyframes have no diagnostics - [see reproduction](https://wixplosives.github.io/stylable-playground/#state=%28%27files%21%281%21%27%2540st-import%2520%255Bkeyframes%257B3%257D%255D%2520from%2520%255C%27.2%255C%2740.root%2520%280%2520%2520%2520%2520animation-name%253A%252034%29%27%7E2%21%27%27%29%7Eselected%211%27%29*.st.css0%255Cn1%27%252Findex*2%252Fother*3unknown4%253B0%250143210*_)
- [x] change current nesting diagnostics that only reports under `@st-scope` to report under any unknown nesting with the exception of `@media` and stylesheet root
- [x] don't report `@keyframes` override under different `@media` rules  ([see demo](https://jsbin.com/gefepofanu/1/edit?html,css,output)) - **keep report on import override or any other collision**.
- [x] fix global keyframes with no name from registering a symbol

## refactor
- [x] move related code into the feature (`KeyframesSymbol`, `reservedKeyFrames`)
- [x] processor - move code to `analyzeAtRule`
- [x] transformer - split keyframes resolve/transform/export into the general flow
  - `transformResolve` (new hook) - move `resolveKeyframes` to feature
  - `transformAtRuleNode` (new hook) - move to feature
  - `transformDeclaration` (new hook) - move `animation/animation-name` namespace to feature
  - `transformJSExports` (new hook) - move to feature
- [x] change `st-import` feature to register keyframes to `st-symbol`
  - [x] make the registration plugable from `css-keyframes`
- [x] align `redeclare` keyframes symbol with other symbol diagnostics through `st-symbol`

## deprecate
- [x] `meta.keyframes`
- [x] `meta.mappedKeyframes`

## tests
- [x] move tests to `features/css-keyframes.spec`
- validate feature tests completeness
  - [x] align tests structure
  - [x] add missing tests
    - check `keyframes` symbol collection
    - check local override of imported symbol
    - check diagnostics report for redeclare keyframes
    - js-mixin with `st-global`, unknown and known keyframes mixed-in

## move to backlog
 - [ ] "re-organizing/merge" feature tests template, because currently it is splitted into `meta(analyze), transform, and diagnostics` which causes a lot of duplications - having a single test-kit tool to check collected analyze data, symbols, and diagnostics + transformations and exports would help to combine specs
 - [ ] refactor `css-class` resolve / exports code into feature
 - [ ] `st-mixin` from JS symbols are not part of the mixed-in validation or transformation - for example `animation-name` in mixin cannot resolve `@keyframes` that are defined in the same mixin ([see reproduction](https://wixplosives.github.io/stylable-playground/#state=%28%27files%21%283%21%27%2540st-import%25207%2520from%25201.%252F7180.root5-st-7in%253A%252078%290%27%7E%252F7.js%21%27module.exports%2520%253D%2520funcA%257B%257D5return5*1%2540keyframes%2520a2%28%2961.x2%28B**1expect24%253Dentry__a16*421a16%29B%298%29%27%29%7Eselected%213%27%29*990%255Cn1%255C%272%255C%21%25203%27%252Findex.st.css41animaA-name5%2520%28B6%252CB*7mix8%253B09%2520%2520AtionB0*%2501BA9876543210*_)) 
- [ ] keyframe escaping bug #1930
